workitem/enum_type_blackbox_test.go

```diff
@@ -321,3 +321,86 @@ func TestEnumType_EqualEnclosing(t *testing.T) {
 		})
 	})
 }
+
+func TestEnumType_ConvertFromModel(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+	type testCase struct {
+		subTestName    string
+		input          interface{} // contains valid and invalid values
+		expectedOutput interface{}
+		wantErr        bool
+	}
+	tests := []struct {
+		name string
+		enum w.EnumType
+		data []testCase
+	}{
+		{
+			"kind string",
+			w.EnumType{
+				SimpleType: w.SimpleType{Kind: w.KindEnum},
+				BaseType:   w.SimpleType{Kind: w.KindString},
+				Values:     []interface{}{"first", "second", "third"},
+			},
+			[]testCase{
+				{"ok", "second", "second", false},
+				{"ok - nil", nil, nil, false},
+				{"fail - invalid string", "fourth", nil, true},
+				{"fail - int", 11, nil, true},
+				{"fail - float", 1.3, nil, true},
+				{"fail - empty string", "", nil, true},
+				{"fail - list", []string{"x", "y"}, nil, true},
+			},
+		},
+		{
+			"kind int",
```

Comment:
Same here, why do you use an enum of ints and call the tests "kind int"?

Reply:
Same as https://github.com/fabric8-services/fabric8-wit/pull/2224#discussion_r223334899